src/core/kernel.ts

```diff
@@ -322,9 +328,13 @@ export class Kernel<
 		if (!resolvingContext) {
 			respond({ error: 'Can not find context for cross-resolving' });
 		} else {
-			const { resolve } = resolvingContext;
-			resolve(request);
-			respond({ message: 'ok' });
+			const { resolve, resolveCallback } = resolvingContext;
+			if (!resolveCallback) {
+				resolve(request);
+				respond({ message: 'ok' });
+			} else {
+				await resolveCallback(request, respond, resolve);
+			}
```

Comment:
I should add a unit test for this one

Reply:
also, how about auto repond `respond({ message: 'ok' });` after `await resolveCallback(request, respond, resolve);`. That means the `resolveCallback` is used to proxy the request to resolve only, don't need to care about respond to resolver. And the middleware itself only throw Error if the resolving request is not accepted